enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v3_2/codegen/GeneratedMethodStructure.scala

```diff
@@ -425,6 +425,12 @@ class GeneratedMethodStructure(val fields: Fields, val generator: CodeBlock, aux
 
   override def nodeGetRelationshipsWithDirection(iterVar: String, nodeVar: String, direction: SemanticDirection) = {
     val local = generator.declare(typeRef[RelationshipIterator], iterVar)
+    generator.assign(local, constant(null))
+    //Make sure cursor is always closed
+    _finalizers.append((_: Boolean) => (block) =>
+      using(block.ifStatement(Expression.notNull(block.load(local.name())))) {inner =>
+        inner.expression(invoke(inner.load(local.name()), method[RelationshipIterator, Unit]("close")))
+      })
```

Comment:
You are correct, nice catch! 

I fixed so that it is called each time it has exited loop, the solution is not super nice. But I am afraid making it nice will require some refactoring which I am not sure we should do at this point in time